server/app/services/program/ProgramServiceImpl.java

```diff
@@ -68,6 +87,71 @@ public ProgramServiceImpl(
     this.versionRepository = checkNotNull(versionRepository);
   }
 
+  /**
+   * Parses the given value based on the given scalar type and operator. For example, if the scalar
+   * is of type LONG and the operator is of type ANY_OF, the value will be parsed as a list of
+   * comma-separated longs.
+   *
+   * <p>If value is the empty string, then parses the list of values instead.
+   */
+  // TODO: make this private once the old predicate logic is removed from
+  // AdminProgramBlockPredicatesController
+  @VisibleForTesting
+  public static PredicateValue parsePredicateValue(
```

Comment:
note: this method moved from AdminProgramBlockPredicatesController but otherwise unchanged

Reply:
done